desktop_version/src/TextInput.cpp

```diff
@@ -0,0 +1,767 @@
+#include <SDL.h>
+#include <algorithm>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <iostream>
+
+#include "KeyPoll.h"
+#include "TextInput.h"
+#include "UTF8.h"
+#include "UtilityClass.h"
+#include "Vlogging.h"
```

Comment:
1. Why `#include` `<algorithm>`, `<sstream>`, and `<iostream>`? Do you ever make use of any of them? If not, then remove them.
2. `#include "TextInput.h"` should be at the very top, separated by a blank line, since this file (`TextInput.cpp`) is the file that implements it. This is how other files in the codebase do it.